src/appleseed/foundation/image/image.h

```diff
@@ -80,6 +80,11 @@ class APPLESEED_DLLSYMBOL Image
         const size_t        tile_height,        // tile height, in pixels
         const PixelFormat   pixel_format);
 
+    // Copy image data but allow to tweak the channel_count    
+    Image(
```

Comment:
I understand the need here but this API is too limited, it only allows to _reduce_ the number of channels. In fact your implementation is even more limited since it asserts that the source image has exactly four channels and the destination one has exactly three channels, which is not reflected in any comment here.

`renderer::Tile` offers a more powerful mechanism with "shuffle tables", I would suggest exposing this mechanism in `renderer::Image` as well.

Reply:
Wow, I get it. I will use that function to do it.